spec/requests/api/connect/v3/systems/products_controller_spec.rb

```diff
@@ -207,90 +207,82 @@
   end
 
   describe '#upgrade' do
+    let(:request) { put url, headers: headers, params: payload }
+    let(:new_product) { FactoryGirl.create(:product, :with_mirrored_repositories) }
+    let(:payload) do
+      {
+        identifier: new_product.identifier,
+        version: new_product.version,
+        arch: new_product.arch
+      }
+    end
+    let(:serialized_json) do
+      V3::ServiceSerializer.new(
+        new_product.service,
+        base_url: URI::HTTP.build({ scheme: response.request.scheme, host: response.request.host }).to_s
+      ).to_json
+    end
+
     it_behaves_like 'products controller action' do
       let(:verb) { 'put' }
     end
 
-    context 'with not activated product' do
-      before { put url, headers: headers, params: payload }
-      subject { response }
+    before do
+      request
+      system.reload
+    end
 
-      let(:product) { FactoryGirl.create(:product, :with_mirrored_repositories) }
-      let(:payload) do
-        {
-          identifier: product.identifier,
-          version: product.version,
-          arch: product.arch
-        }
-      end
+    context 'new product' do
+      subject { response }
 
-      let(:error_response) do
-        {
-          type: 'error',
-          error: "No activation with product '#{product.friendly_name}' was found.",
-          localized_error: "No activation with product '#{product.friendly_name}' was found."
-        }
-      end
+      specify { expect(system.activations.count).to eq(1) }
```

Comment:
1. Didn't know that `specify` was a thing, TIL. :-) It is used in very few places, maybe we should just use `it` instead?
2. The example should have a name, otherwise in the `--format=documentation` it looks like this:
```
#upgrade
  new product
    should eq 1
```

Reply:
I didn't know about `specify` either, it was in the specs there already. I will see about addressing the other issues.

Reply:
If it was in the specs already doesn't mean it should stay!

Reply:
Used only 19 times in glue. Looks like really old stuff :grimacing:

Reply:
True, but I didn't have a problem with it other than the documentation issue that you had.

Reply:
@tmuntaner @ikapelyukhin I've been sprinkling `specify`s in our tests, sorry hehe. (It's not "really old stuff). But I can explain.

when you have a `subject` you can then say `it { is_expected ... }`, which looks nice.
But sometimes you need to run the subject within a block, to check for example if it raises an error: `it { expect { subject }.to raise_error }`
But "it expect" looks weird, so in such a case, `specify` is used: `specify { expect { subject }.to raise_error }`

In this particular case, I would agree that `specify` doesn't really need to be used, since there's no mention of `subject`. And probably, a more descriptive `it` block would be better.